### DISCUSSION TITLE
Questions about propane mechanism / addition of large mechanism

### DISCUSSION THREAD

**Discussion by PBernigaud** [AMReX-Combustion/PeleC#763] [Q&A] [answered]
Dear all, 

I am currently trying to simulate the combustion of propane. To this end, I used the kinetic mechanism propane_fc included in PelePhysics. As a first question, what is the reference for this model?  

I am now trying to use the kinetic mechanism developed at UC San Diego (https://web.eng.ucsd.edu/mae/groups/combustion/mechanism.html), but I am encountering difficulties when building the exec file.  When trying to compile the file ReactorUtils from PelePhysics, the build process freezes while memory usage significantly increases (up to 40 Go).  Letting run the build process for 1+ hour doesn't solve the problem. It appears that artificially reducing the number of reactions in the UCSD mechanism from around 250 to 190 makes things run smoothly.
Do you see any reason for the number of reactions in the mechanism to influence the build process? Do you know of a get-around to use large mechanisms with PeleC? 

I am thanking you in advance for your help.

**Comment by drummerdoc** [comment]
What computer system are you working on when this happens?

**Reply by PBernigaud** [reply]
I am working on a HPC super-computer, compiling with Intel 23. Which informations would you need more precisely?

**Reply by drummerdoc** [reply]
Specifically, I wanted to know if you were building for GPU or CPU, and if GPU, which type.

**Reply by PBernigaud** [reply]
I am building for CPU.

**Reply by jrood-nrel** [reply]
If compiling for Intel GPUs using OneAPI/SyCL with AOT, one can expect compiling to take all day for large mechanisms. If using the "classic" Intel compilers for the CPU, I'm guessing it should be less. The Intel optimizer will likely spend a long time on these huge files though. You could also try compiling with something like `-O2` instead.

**Comment by baperry2** [comment]
Looks like the propane_fc mechanism was added in [this commit](https://github.com/AMReX-Combustion/PelePhysics/commit/d28db5e463c2136a53da5bc90ef85012753874e2), which references [this paper](https://www.sciencedirect.com/science/article/pii/S0010218018300208)

**Comment by esclapez** [comment, marked as answer]
This propane mechanism was derived by Felix Collin (hence FC). It's available here:
https://chemistry.cerfacs.fr/en/chemical-database/mechanisms-list/c3h8_34_173_fc-mechanism/

Note that a QSS version (with 12 species in QSS) is also available on that link but not available in PelePhysics at this time.

**Comment by PBernigaud** [comment]
Dear all, 

Thank you for your helpful answers, and sorry for the delay. 
The paper is indeed the one I was looking for. 
Regarding the compilation issue, it appears to be caused by the Intel 23 C++ compiler. Using Intel 22 solved the problem. 

Thanks again !
